src/components/SourcesModal.js

```diff
@@ -0,0 +1,208 @@
+// @flow
+
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import { filter } from "fuzzaldrin-plus";
+
+import actions from "../actions";
+import {
+  getSources,
+  getActiveSearch,
+  getSourceSearchQuery
+} from "../selectors";
+import { endTruncateStr } from "../utils/utils";
+import { scrollList } from "../utils/result-list";
+import { isPretty, getSourcePath, isThirdParty } from "../utils/source";
+import Modal from "./shared/Modal";
+import SearchInput from "./shared/SearchInput";
+import ResultList from "./shared/ResultList";
+
+import type { SourcesMap } from "../reducers/sources";
+
+function formatResults(sources: SourcesMap) {
+  return sources
+    .valueSeq()
+    .toJS()
+    .filter(source => !isPretty(source) && !isThirdParty(source))
+    .map(source => ({
+      value: getSourcePath(source),
+      title: getSourcePath(source)
+        .split("/")
+        .pop(),
+      subtitle: endTruncateStr(getSourcePath(source), 100),
+      id: source.id
+    }))
+    .filter(formattedSource => formattedSource.value != "");
```

Comment:
Some sources have no value or title and I filter them out but I think we should do something better. This would make a good follow-up issue.